### DISCUSSION TITLE
Video files in liquidsoap: stream does not start yet

### DISCUSSION THREAD

**Discussion by suwofis** [savonet/liquidsoap#4115] [User Support]
Thanks to your great help, I managed to get a liquidsoap radiostream via icecast started. Now I tried the same wth a video file. My file looked like this: `s = single("test.mp4")

output.icecast(
  %theora,
  host = "localhost",
  port = 8000,
  password = "hackme",
  mount = "test.mp4",
  s
)`

Liquidsoap didn't play the file and ended up giving out the following log: `2024/08/27 00:44:54 >>> LOG START
2024/08/27 00:44:43 [ffmpeg.filter.bitstream:3] No valid mode found for filter pgs_frame_merge!
2024/08/27 00:44:43 [main:3] Liquidsoap 2.1.3
2024/08/27 00:44:43 [main:3] Using: graphics=5.1.2 bytes=[distributed with OCaml 4.02 or above] pcre=7.5.0 sedlex=3.0 menhirLib=20220210 curl=0.9.2 uri=4.2.0 dtools=0.4.4 duppy=0.9.2 cry=0.6.7 mm=0.8.1 xmlplaylist=0.1.5 lastfm=0.3.3 ogg=0.7.3 ogg.decoder=0.7.3 vorbis=0.8.0 vorbis.decoder=0.8.0 opus=0.2.2 opus.decoder=0.2.2 speex=0.4.1 speex.decoder=0.4.1 mad=0.5.2 flac=0.3.1 flac.ogg=0.3.1 flac.decoder=0.3.1 dynlink=[distributed with Ocaml] lame=0.3.6 shine=0.2.3 gstreamer=0.3.1 frei0r=0.1.2 theora=0.4.0 theora.decoder=0.4.0 ffmpeg=1.1.6 bjack=0.1.6 alsa=0.3.0 ao=0.2.4 samplerate=0.1.6 taglib=0.3.10 ssl=0.5.11 magic=0.7.3 camomile=1.0.2 inotify=2.4.1 faad=0.5.1 soundtouch=0.1.9 portaudio=0.2.3 pulseaudio=0.1.5 ladspa=0.2.2 camlimages=4.2.6 camlimages.freetype=5.0.4 lo=0.2.0 gd=1.0a5
2024/08/27 00:44:43 [dynamic.loader:3] Could not find dynamic module for fdkaac encoder.
2024/08/27 00:44:43 [gstreamer.loader:3] Loaded GStreamer 1.22.0 0
2024/08/27 00:44:43 [clock:3] Using builtin (low-precision) implementation for latency control
2024/08/27 00:44:54 [frame:3] Using 44100Hz audio, 25Hz video, 44100Hz main.
2024/08/27 00:44:54 [frame:3] Video frame size set to: 1280x720
2024/08/27 00:44:54 [frame:3] Frame size must be a multiple of 1764 ticks = 1764 audio samples = 1 video samples.
2024/08/27 00:44:54 [frame:3] Targeting 'frame.duration': 0.04s = 1764 audio samples = 1764 ticks.
2024/08/27 00:44:54 [frame:3] Frames last 0.04s = 1764 audio samples = 1 video samples = 1764 ticks.
2024/08/27 00:44:54 [sandbox:3] Sandboxing disabled
2024/08/27 00:44:54 [video.converter:3] Using preferred video converter: ffmpeg.
2024/08/27 00:44:54 [audio.converter:3] Using samplerate converter: libsamplerate.
2024/08/27 00:44:54 [single_0:3] "test.mp4" is static, resolving once for all...
2024/08/27 00:44:55 [test.mp4:3] Prepared "test.mp4" (RID 0).
2024/08/27 00:44:55 [clock.main:3] Streaming loop starts in auto-sync mode
2024/08/27 00:44:55 [video.text:3] Using camlimages implementation
2024/08/27 00:44:55 [clock.main:3] Delegating synchronisation to CPU clock
2024/08/27 00:44:55 [test.mp4:3] Connecting mount test.mp4 for source@localhost...
2024/08/27 00:44:55 [test.mp4:3] Connection setup was successful.
2024/08/27 00:44:55 [clock.main:2] Source test.mp4 failed while streaming: Theora.Invalid_data!
2024/08/27 00:44:55 [test.mp4:3] Closing connection...
2024/08/27 00:44:55 [clock.main:3] Streaming loop stopped.
2024/08/27 00:44:55 [main:3] Shutdown started!
2024/08/27 00:44:55 [main:3] Waiting for main threads to terminate...
2024/08/27 00:44:55 [main:3] Main threads terminated.
2024/08/27 00:44:55 [threads:3] Shutting down scheduler...
2024/08/27 00:44:55 [threads:3] Scheduler shut down.
2024/08/27 00:44:55 [main:3] Cleaning downloaded files...
2024/08/27 00:44:55 [main:3] Freeing memory...
`
If I use %mp4 instead of %theora it says that the format is unsupported... Any ideas what I can do?

Thanks in advance

**Comment by vitoyucepi** [comment]
Hi @suwofis,
It seems that the `%theora` encoder is broken, so you can't use it. Ref: #4116.
Consider using the `%ffmpeg` encoder instead.
```ruby
output.icecast(
  %ffmpeg(format="ogg",
    %audio(codec="libvorbis"),
    %video(codec="libtheora")
  ),
  s,
  host = "localhost",
  port = 8000,
  password = "hackme",
  mount = "test.mp4",
)
```

**Reply by suwofis** [reply]
Ok, thank you very much for your reply.
